### PR TITLE
table `pull_request` wasn't updated correctly

### PR DESCRIPTION
**Pull request by kaz**
Since this commit (https://github.com/go-gitea/gitea/commit/dd55534b82ca9527b68fddf0f63ebb13c105c466), table `pull_request` wasn't updated correctly.

Because of this, merge description says such wrong thing:
**Ghost** merged 1 commits from user/branch into user/branch **48 years ago**

Example https://try.gitea.io/amama/haipro/pulls/1

### CODE REVIEW

**Comment by ethantkoenig** [comment]
LGTM

File: models/pull.go [hunk 1]
```diff
@@ -422,7 +422,7 @@ func (pr *PullRequest) setMerged() (err error) {
 	if err = pr.Issue.changeStatus(sess, pr.Merger, pr.Issue.Repo, true); err != nil {
 		return fmt.Errorf("Issue.changeStatus: %v", err)
 	}
-	if _, err = sess.Id(pr.ID).Cols("has_merged").Update(pr); err != nil {
+	if _, err = sess.Id(pr.ID).Cols("has_merged, merged_commit_id, merger_id, merged_unix").Update(pr); err != nil {
```
Comment (daviian) [inline models/pull.go]: Shouldn't it be `sess.ID` ?

Comment (lafriks) [reply models/pull.go]: They are aliases but by GoLang naming ID is more correct

Comment (ethantkoenig) [reply models/pull.go]: `Id(..)` is deprecated, so I think we should prefer `ID(..)` (https://godoc.org/github.com/go-xorm/xorm#Engine.Id)

Comment (kaz) [reply models/pull.go]: Umm ... but `Id(..)` is used in many other places.
It seems to be reasonable to fix all of them at once as an other issue.

Comment (lafriks) [reply models/pull.go]: To not to accidently break something it's better to change them one by one when touching old code like in this case

Comment (kaz) [reply models/pull.go]: Okay, I'll fix it

**Comment by tonivj5** [comment]
https://try.gitea.io/amama/haipro/pulls/1 Why if do I enter into commits or files it shows a 500 error? :confused:

**Comment by kaz** [comment]
@xxxtonixxx 

It is because `merged_commit_id` column won't be updated.
Due to this, `GetCommit()` and `GetDiffRange()` fail trying to open the commit with empty id.

This PR fixes this bug too.

**Comment by kaz** [comment]
rebased

**Comment by lafriks** [comment]
LGTM

**Pull request merged**